packages/swayze/lib/src/widgets/table_body/selections/fill_selections/fill_selection.dart

```diff
@@ -0,0 +1,270 @@
+    ) as Tween<double>?;
+
+    _top = visitor(
+      _top,
+      widget.offset.dy,
+      (dynamic value) => Tween<double>(begin: value as double),
+    ) as Tween<double>?;
+
+    _size = visitor(
+      _size,
+      widget.size,
+      (dynamic value) => SizeTween(begin: value as Size),
+    ) as SizeTween?;
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    final animation = this.animation;
+    final left = _left?.evaluate(animation) ?? 0;
+    final top = _top?.evaluate(animation) ?? 0;
+    final size = _size?.evaluate(animation) ?? Size.zero;
+
+    return FillSelectionPainter(
+      border: widget.border,
+      offset: Offset(left, top),
+      size: size,
+    );
+  }
+}
+
+/// A [LeafRenderObjectWidget] that render a fill selection.
+@visibleForTesting
+class FillSelectionPainter extends LeafRenderObjectWidget {
+  final SelectionBorder border;
+  final Offset offset;
+  final Size size;
+
+  const FillSelectionPainter({
+    required this.border,
+    required this.offset,
+    required this.size,
+  });
+
+  @override
+  _RenderFillSelectionPainter createRenderObject(BuildContext context) {
+    return _RenderFillSelectionPainter(
+      border,
+      offset,
+      size,
+    );
+  }
+
+  @override
+  void updateRenderObject(
+    BuildContext context,
+    _RenderFillSelectionPainter renderObject,
+  ) {
+    renderObject
+      ..border = border
+      ..offset = offset
+      ..definedSize = size;
+  }
+}
+
+class _RenderFillSelectionPainter extends RenderBox {
+  _RenderFillSelectionPainter(
+    this._border,
+    this._offset,
+    this._definedSize,
+  );
+
+  Offset _offset;
+  Offset get offset => _offset;
+  set offset(Offset value) {
+    if (_offset != value) {
+      _offset = value;
+      markNeedsPaint();
+    }
+  }
+
+  Size _definedSize;
+  Size get definedSize => _definedSize;
+  set definedSize(Size value) {
+    if (_definedSize != value) {
+      _definedSize = value;
+      markNeedsLayout();
+    }
+  }
+
+  SelectionBorder _border;
+  SelectionBorder get border => _border;
+  set border(SelectionBorder value) {
+    if (_border != value) {
+      _border = value;
+      markNeedsPaint();
+    }
+  }
+
+  @override
+  void performLayout() {
+    size = constraints.constrain(_definedSize);
+  }
+
+  @override
+  void paint(PaintingContext context, Offset offset) {
+    final canvas = context.canvas;
+
+    canvas.save();
+    canvas.translate(offset.dx, offset.dy);
+
+    paintSelectionBorder(
+      canvas,
+      _offset & size,
+      border,
+    );
+
+    canvas.restore();
+  }
+}
+
+extension on UserSelectionModel {
+  SelectionStyle? toSelectionStyle(BuildContext context) {
+    final model = this;
+
+    return style ??
+        (model is CellUserSelectionModel
+            ? model.type.toSelectionStyle(context)
+            : null);
+  }
+}
+
+extension on CellUserSelectionType {
+  SelectionStyle? toSelectionStyle(BuildContext context) {
+    switch (this) {
```

Comment:
In both these extensions, may an early exist improves readability:]

```dart
if (style != null) {
   return style;
}

final model = this;
if (model is! CellUserSelectionModel) {
   return null;
}

return model.type.toSelectionStyle(context);
```


and 

```dart
if (this is! CellUserSelectionType.fill) {
  return null;
}

final style = InternalScope.of(context).style;

return SelectionStyle.dashedBorderOnly(
  color: style.dragAndFillStyle.color,
  borderWidth: style.dragAndFillStyle.borderWidth,
);
```